Add detailed support for Airwell A/C protocol.

crankyoldgit opened the pull request:
* Add support for controling:
  - Operation Mode
  - Power Toggle
  - Fan Speed
  - Temperature
* Common A/C API support added.
* Unit tests

Fixes #1202

crankyoldgit committed:
Add detailed support for Airwell A/C protocol.
* Add support for controling:
  - Operation Mode
  - Power Toggle
  - Fan Speed
  - Temperature
* Common A/C API support added.
* Unit tests

Fixes #1202

juanbaldovi commented:
Hello
I've flashed IRMQTTServer and made some quick tests and it seems to work perfectly.
I'll make some more tests later this week.
I'll keep you updated.
Thanks for all your support!

crankyoldgit commented:
@juanbaldovi Thanks for confirming it's working! That helps!

NiKiZe commented on test/ir_Airwell_test.cpp:
```diff
@@ -250,4 +251,142 @@ TEST(TestDecodeAirwell, RealExample3) {
   EXPECT_EQ(0x60080002, irsend.capture.value);
   EXPECT_EQ(0x0, irsend.capture.address);
   EXPECT_EQ(0x0, irsend.capture.command);
+  EXPECT_EQ(
```
This is the only testdata that is valid for this AC? If not should it be added to the above as well.

And if the other data also is an ac, should there be models added? (even if not I just want to make a mental note about it)

crankyoldgit replied:
> This is the only testdata that is valid for this AC? If not should it be added to the above as well.

No. Yes, added/fixed. Yeah, I was lazy.

> And if the other data also is an ac, should there be models added? (even if not I just want to make a mental note about it)

It appears it's all a single common "protocol" based on comparing the comments in the Original issue #1069 describing the raw captures and what the library decodes them to.

crankyoldgit committed:
Enforce a fan speed of Low when in Dry mode.
Per https://docs.google.com/spreadsheets/d/1MeqVsgXQAMSiAx7zNunG8qbFcIT8tqZtcjLmMa0GEng/edit#gid=0&range=A42:F45
For #1202

crankyoldgit committed:
Code review feedback
- Add result to string output for earlier real examples.
